static/client/world.js

```diff
@@ -173,10 +173,12 @@ CCC.World.receiveMessage = function(e) {
  */
 CCC.World.preprocessMessage = function(msg) {
   if (Array.isArray(msg)) {
-    for (var i = 0; i < msg.length; i++) {
-      CCC.World.preprocessMessage(msg[i]);
+    for (var singleMsg of msg) {
```

Comment:
I thought `singleMsg` was a big ugly, and wondered, given the new block scoping rules as well as the order that things get evaluated in, if it would be legal to write:

```JS
    for (const msg of msg) {
```

…where the first `msg` was a new const binding scoped to the for-of, while the second one was the existing var binding.  But (perhaps fortunately!) this isn't legal: the second `msg` is in fact the same block-scoped const binding as the first, which (since the it has not yet been initialised) is in the temporal dead zone and so you'd get `ReferenceError: msg is not defined`.

I am somewhat relieved.